pkg/logs/input/tailer/tailer_nix.go

```diff
@@ -62,8 +62,8 @@ func (t *Tailer) readForever() {
 				t.wait()
 				continue
 			}
-			t.decoder.InputChan <- decoder.NewInput(inBuf[:n])
 			t.incrementReadOffset(n)
+			t.decoder.InputChan <- decoder.NewInput(inBuf[:n])
```

Comment:
why are we doing this change? I find it dangerous to increment the read offset before we passed the data around

Reply:
I thought this would fix some bugs, I'll revert.